R/utilities-figures.R

```diff
@@ -223,3 +223,160 @@ createEsqlabsExportConfiguration <- function(projectConfiguration) {
 
   return(exportConfiguration)
 }
+
+#' Generate plots as defined in excel file `projectConfiguration$plotsFile`
+#'
+#' @param simulatedScenarios A list of simulated scenarios as returned by `runScenarios()`
+#' @param observedData A list of `DataSet` objects
+#' @param projectConfiguration Object of class `ProjectConfiguration`
+#' that contains information about the output paths and the excel file
+#' where plots are defined.
+#' @param stopIfNotFound If TRUE (default), the function stops if any of the
+#' simulated results or observed data are not found. If FALSE a warning is printed.
+#'
+#'
+#' @return A list of `ggplot` objects
+#'
+#' @export
+createPlotsFromExcel <- function(simulatedScenarios, observedData, projectConfiguration, stopIfNotFound = TRUE) {
+  validateIsOfType(observedData, "DataSet", nullAllowed = TRUE)
+  validateIsOfType(projectConfiguration, "ProjectConfiguration")
+  dfDataCombined <- readExcel(projectConfiguration$plotsFile, sheet = "DataCombined")
+
+  # warnings for invalid data in plot definitions from excel
+  # scenario not present in simulatedScenarios
+  missingScenarios <- setdiff(setdiff(dfDataCombined$scenario, names(simulatedScenarios)), NA)
+  if (length(missingScenarios) != 0) {
+    if (stopIfNotFound) {error(messages$stopInvalidScenarioName(missingScenarios))}
+    warning(messages$warningInvalidScenarioName(missingScenarios, projectConfiguration$plotsFile))
+    dfDataCombined <- dfDataCombined[!(dfDataCombined$scenario %in% missingScenarios), ]
+  }
+  # data set name not present in observedData
+  missingDataSets <- setdiff(setdiff(dfDataCombined$dataSet, names(observedData)), NA)
+  if (length(missingDataSets) != 0) {
+    if (stopIfNotFound) {error(messages$stopInvalidDataSetName(missingDataSets))}
```

Comment:
`error`, had too much Python coding? :D It should be `stop` ;) (or does it really work with `error`?)

Reply:
Of course, produces an error as intended ;)